queue/bytes_queue.go

```diff
@@ -78,6 +78,7 @@ func (q *BytesQueue) Reset() {
 	q.head = leftMarginIndex
 	q.rightMargin = leftMarginIndex
 	q.count = 0
+	q.full = false
```

Comment:
Can you add a test for this?

Reply:
I'd like to, but what kind of test do you mean? 
I think it's kind of obvious that without resetting this flag, the queue will still be full after ``Reset()``, and the next ``Push()`` will cause a re-allocation on this empty queue

Reply:
Exactly,